cli/server.go

```diff
@@ -1406,15 +1406,27 @@ func configureTLS(tlsMinVersion, tlsClientAuth string, tlsCertFiles, tlsKeyFiles
 		return nil, nil //nolint:nilnil
 	}
 
-	err = configureCAPool(tlsClientCAFile, tlsConfig)
+	err = configureClientCAPool(tlsClientCAFile, tlsConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	return tlsConfig, nil
 }
 
-func configureCAPool(tlsClientCAFile string, tlsConfig *tls.Config) error {
+func configureRootCAPool(tlsConfig *tls.Config) error {
+	// Note that there is no way to reload certificates so if we need that we
+	// would have to copy the implementation although there is different behavior
+	// on different operating systems so it could be a lot.
```

Comment:
This is actually making it quite difficult to test.  Should I just copy the whole implementation instead?  I know in v1 we basically did that but it was only for Linux and the standard library handles more than that.

I am also not entirely sure how to test manually, either but this client seems to be used in OIDC so maybe if I set up a provider with a self-signed certificate I can reproduce that way.